### PR TITLE
SpyCloud Enterprise Protection Initial Commit

### PR DESCRIPTION
**Pull request by loginsoft-integrations**
Required items, please complete
   
   Change(s):
   - See guidance below

   Reason for Change(s):
   - See guidance below

   Version Updated:
   - Required only for Detections/Analytic Rule templates
   - See guidance below

   Testing Completed:
   - See guidance below

   Checked that the validations are passing and have addressed any issues that are present:
   - See guidance below

# Guidance <- remove section before submitting
-----------------------------------------------------------------------------------------------------------
## Before submitting this PR please ensure that you have read the following sections and filled out the changes, reason for change and testing complete sections:

Thank you for your contribution to the Microsoft Sentinel Github repo.

> Details of the code changes in your submitted PR.  Providing descriptions for pull requests ensures there is context to changes being made and greatly enhances the code review process.  Providing associated Issues that this resolves also easily connects the reason.
   
   Change(s):
   - Updated syntax for XYZ.yaml

   Reason for Change(s):
   - New schema used for XYZ.yaml
   - Resolves ISSUE #1234

   Version updated:
   - Yes
   - Detections/Analytic Rule templates are required to have the version updated

> The code should have been tested in a Microsoft Sentinel environment that does not have any custom parsers, functions or tables, so that you validate no incorrect syntax and execution functions properly.  If your submission requires a custom parser or function, it must be submitted with the PR.

   Testing Completed:
   - Yes/No/Need Help

_Note: If updating a detection, you must update the version field._

> Before the submission has been made, please look at running the KQL and Yaml Validation Checks locally.
> https://github.com/Azure/Azure-Sentinel#run-kql-validation-locally

   Checked that the validations are passing and have addressed any issues that are present:
   - Yes/No/Need Help
   
   _Note: Let us know if you have tried fixing the validation error and need help._

> References: 
> - [Guidance for Detection checks](https://github.com/Azure/Azure-Sentinel#pull-request-detection-template-structure-validation-check)
> - [General contribution guidance](https://github.com/Azure/Azure-Sentinel/wiki#what-can-you-contribute-and-how-can-you-create-contributions)
> - [PR validation troubleshooting](https://github.com/Azure/Azure-Sentinel#pull-request)

-----------------------------------------------------------------------------------------------------------

### CODE REVIEW

**Comment by RamboV** [comment]
> Hello @loginsoft-integrations as per latest guidelines issued by Microsoft Sentinel, please repackage the solution using V3 tool Please go through this readme file on how to use the V3 tool https://github.com/Azure/Azure-Sentinel/blob/master/Tools/Create-Azure-Sentinel-Solution/V3/README.md

Hello @v-sudkharat ,

Sure, Once the files are approved we will start the solution package process

**Comment by v-sudkharat** [comment]
@RamboV Ok, we will review the content first and share our feedback, you can expect our review update by 26Jul 2023. I can see that analytical rules uploaded in .json format but we need rules in .yaml format and also attach sample data.

**Comment by v-sudkharat** [comment]
Hello @loginsoft-integrations, as per latest guidelines issued by Microsoft Sentinel, please repackage the solution using "V3 tool." and add "Release Notes" for solution.
Please read this readme file for instructions on using the "V3 tool" and while creating "Release Notes".

V3 Tool Link: - `https://github.com/Azure/Azure-Sentinel/blob/master/Tools/Create-Azure-Sentinel-Solution/V3/README.md`
Release Notes Link: -` https://github.com/Azure/Azure-Sentinel/blob/master/Solutions/ReleaseNotesGuidance.md`

We are working on reviewing the files. Thanks

**Comment by v-sudkharat** [comment]
Hi @loginsoft-integrations, hope you are doing well. Just wanted to check if you got a chance to look at the suggestions shared. Please feel free to reach out to us for any queries or support.

**Comment by RamboV** [comment]
> Hi @loginsoft-integrations, hope you are doing well. Just wanted to check if you got a chance to look at the suggestions shared. Please feel free to reach out to us for any queries or support.

Hello, Thanks. Yeah we looked into the links shared. Once the PR is approved we will be working on packaging .

**Comment by v-sudkharat** [comment]
Hi @RamboV, As requested earlier, please can you add sample data to this location : [Sample Data/Custom](https://github.com/Azure/Azure-Sentinel/tree/master/Sample%20Data/Custom)

**Comment by v-sudkharat** [comment]
Hi @RamboV, thank you for accommodating the suggested changes! We will review the Pull Request and get back to you by 1st Aug 2023.

**Comment by v-sudkharat** [comment]
Hi @loginsoft-integrations, please update the sample data file name, it should be same as table name used in query and please update tactics and techniques in analytic rule from this guide: -` https://attack.mitre.org/versions/v11/matrices/enterprise/`. Thanks!

File: Solutions/SpyCloud Enterprise Protection/Analytics Rules/SpyCloudEnterpriseProtectionBreachRule.yaml
Comment (v-sudkharat) [inline Solutions/SpyCloud Enterprise Protection/Analytics Rules/SpyCloudEnterpriseProtectionBreachRule.yaml:28]: Please remove "groupByCustomDetails: []" in both analytic rules, if you are not using it.

File: Sample Data/Custom/SpyCloud_Watchlist_Data_CL.json
Comment (v-sudkharat) [inline Sample Data/Custom/SpyCloud_Watchlist_Data_CL.json:5]: Please update the value for email, it must be sanitized@sanitized.com at all places.

**Comment by v-sudkharat** [comment]
Hi @loginsoft-integrations, hope you are doing well. Just wanted to check if you got a chance to look at the suggestions shared. Please feel free to reach out to us for any queries or support. Thanks!

**Comment by RamboV** [comment]
> Hi @loginsoft-integrations, hope you are doing well. Just wanted to check if you got a chance to look at the suggestions shared. Please feel free to reach out to us for any queries or support. Thanks!

Hello @v-sudkharat we are looking into them will update tomorrow.  Thanks.

**Comment by v-sudkharat** [comment]
@RamboV, Ok. Thanks!

**Comment by RamboV** [comment]
> Hi @loginsoft-integrations, please update the sample data file name, it should be same as table name used in query and please update tactics and techniques in analytic rule from this guide: -` https://attack.mitre.org/versions/v11/matrices/enterprise/`. Thanks!

Updated

File: Solutions/SpyCloud Enterprise Protection/Analytics Rules/SpyCloudEnterpriseProtectionBreachRule.yaml
Comment (RamboV) [reply Solutions/SpyCloud Enterprise Protection/Analytics Rules/SpyCloudEnterpriseProtectionBreachRule.yaml]: Removed

File: Sample Data/Custom/SpyCloud_Watchlist_Data_CL.json
Comment (RamboV) [reply Sample Data/Custom/SpyCloud_Watchlist_Data_CL.json]: Updated

**Comment by RamboV** [comment]
Hello @v-sudkharat , Sudhakar I have attended all the suggestions kindly review and provide your feedback

**Comment by v-sudkharat** [comment]
Hi @RamboV, thank you for accommodating the suggested changes! We will review the Pull Request and get back to you by 09-08-2023. Thanks!

**Comment by v-sudkharat** [comment]
Hi @RamboV, we successfully deployed the playbook, but we are not authorized to run successful playbook till end, so please can you attach the Successfully deployed and running playbook screenshot in this chat. 
In analytical rules please check and correct the indentation as per following sample analytical rule - https://github.com/Azure/Azure-Sentinel/blob/master/Solutions/Azure%20Active%20Directory/Analytic%20Rules/nrt_FirstAppOrServicePrincipalCredential.yaml
image

**Comment by v-sudkharat** [comment]
Hi @loginsoft-integrations, we have rectified the validation error for analytical rules and sharing the updated files in this chat.
Kindly refer this file and commit the changes or update this file with existing once.
[Analytics Rules.zip](https://github.com/Azure/Azure-Sentinel/files/12395270/Analytics.Rules.zip)
The "Custom Connector" folder should be added into the "Playbooks" folder.
Sharing screenshot for reference-   
image
Please repackage the solution according to the instructions we've already provided. Thanks!
`https://github.com/Azure/Azure-Sentinel/pull/8493#issuecomment-1651586354`

**Comment by RamboV** [comment]
> Hi @loginsoft-integrations, we have rectified the validation error for analytical rules and sharing the updated files in this chat. Kindly refer this file and commit the changes or update this file with existing once. [Analytics Rules.zip](https://github.com/Azure/Azure-Sentinel/files/12395270/Analytics.Rules.zip) The "Custom Connector" folder should be added into the "Playbooks" folder. Sharing screenshot for reference- image Please repackage the solution according to the instructions we've already provided. Thanks! `https://github.com/Azure/Azure-Sentinel/pull/8493#issuecomment-1651586354`

Hi @v-sudkharat , I have attended the changes, once you approve I will start the packaging process.

**Comment by v-atulyadav** [comment]
Hi @loginsoft-integrations,
Thank you for committing these changes. We will review them and get back to you by 25 Aug 2023.

**Comment by v-sudkharat** [comment]
Hi @RamboV, we haven't received the successfully deployed and running playbook screenshot in this chat. please add those screenshots. Other changes look good Please repackage the solution according to the instructions we've already provided.
`https://github.com/Azure/Azure-Sentinel/pull/8493#issuecomment-1651586354 `
Thanks!

**Comment by v-sudkharat** [comment]
yes @RamboV, please add the screenshot of successful running of the playbooks.

**Comment by v-sudkharat** [comment]
Hi @loginsoft-integrations, hope you are doing well. Just wanted to check if you got a chance to look at the suggestions shared. Please feel free to reach out to us for any queries or support. Thanks!

**Comment by v-sudkharat** [comment]
Hi @loginsoft-integrations, we are waiting for your response on above comment. Thanks

**Comment by v-sudkharat** [comment]
Hi @loginsoft-integrations, can you please add the screenshot of successful running of the playbooks. Thanks

**Comment by RamboV** [comment]
> Hi @loginsoft-integrations, can you please add the screenshot of successful running of the playbooks. Thanks

Hi @v-sudkharat , working on it, will send the screenshots by eod, so after the pull is merged than we need to start the packaging process or now?

**Comment by RamboV** [comment]
Hello @v-sudkharat ,

PFA attached screenshots of successful runs of the playbooks. once the PR is merged we will start the packaging process.[
[LogicAppScreenshots.zip](https://github.com/Azure/Azure-Sentinel/files/12512093/LogicAppScreenshots.zip)
](url)

**Comment by v-sudkharat** [comment]
Hi @RamboV, can you please act/respond on above comment? If you have any further questions or need assistance, feel free to reach out to us. Thanks!

**Comment by RamboV** [comment]
> Hi @RamboV, can you please act/respond on above comment? If you have any further questions or need assistance, feel free to reach out to us. Thanks!

Hello @v-sudkharat we are in progress with V3 packaging facing few issues doing so will update you on this tomorrow.

**Comment by RamboV** [comment]
> > Hi @RamboV, can you please act/respond on above comment? If you have any further questions or need assistance, feel free to reach out to us. Thanks!
> 
> Hello @v-sudkharat we are in progress with V3 packaging facing few issues doing so will update you on this tomorrow.

Is it possible to discuss via a call ?

**Comment by v-sudkharat** [comment]
Hi @RamboV, if you are facing issue while repackaging, we can discuss via a call on 08-09-2023. Thanks

**Comment by RamboV** [comment]
> Hi @RamboV, if you are facing issue while repackaging, we can discuss via a call on 08-09-2023. Thanks

Thanks @v-sudkharat , can you please send an invite to vrambatza@loginsoft.com

**Comment by v-sudkharat** [comment]
@RamboV, ok we will send an invite. Thanks

**Comment by RamboV** [comment]
Hello @v-sudkharat , Modified the files according to V3 tooling, please review

**Comment by v-sudkharat** [comment]
Hi @RamboV, as per our discussion in teams chat, please check the zip file which sent by "Sanmit" via a mail and commit the same. Thanks

**Comment by RamboV** [comment]
Hello @v-sudkharat  uploaded the new package please check

File: Solutions/SpyCloud Enterprise Protection/Analytic Rules/SpyCloudEnterpriseProtectionBreachRule.yaml
Comment (v-sudkharat) [inline Solutions/SpyCloud Enterprise Protection/Analytic Rules/SpyCloudEnterpriseProtectionBreachRule.yaml:13]: please remove space in "Credential Access".

File: Solutions/SpyCloud Enterprise Protection/Analytic Rules/SpyCloudEnterpriseProtectionMalwareRule.yaml
Comment (v-sudkharat) [inline Solutions/SpyCloud Enterprise Protection/Analytic Rules/SpyCloudEnterpriseProtectionMalwareRule.yaml:13]: please remove space in "Credential Access".

**Comment by v-sudkharat** [comment]
Hi @RamboV, 
1. please remove the "Sample Data" folder which is inside of Solution folder. 
Sharing screenshot for reference - 
image

2. Add table schema in custom table folder and file name should be same as table name.
https://github.com/Azure/Azure-Sentinel/tree/706e6723f14e72be86f3043b8cbd7dfd8a5719fc/.script/tests/KqlvalidationsTests/CustomTables

**Comment by RamboV** [comment]
Hello @v-sudkharat  I have done modifications as suggested kindly review

File: Solutions/SpyCloud Enterprise Protection/Analytic Rules/SpyCloudEnterpriseProtectionBreachRule.yaml
Comment (v-sudkharat) [inline Solutions/SpyCloud Enterprise Protection/Analytic Rules/SpyCloudEnterpriseProtectionBreachRule.yaml:7]: please add the - "requiredDataConnectors: []" property in both the analytic rules. 
Sharing Screenshot for reference - 
image

**Comment by v-sudkharat** [comment]
Hi @RamboV, thank you for accommodating the suggested changes! We will review the Pull Request.

**Comment by v-sudkharat** [comment]
Hi @RamboV, we added the "Release Notes" link/path in "createUiDefinition" file. please can you update or refer below shared file for existing one. also update that change in "zip" file as well.  Thanks!
[createUiDefinition.zip](https://github.com/Azure/Azure-Sentinel/files/12595144/createUiDefinition.zip)

**Comment by v-sudkharat** [comment]
Hello @RamboV, this solution has multiple playbooks, and we are experiencing an issue while playbook deployment for few playbooks and we are checking this issue, but it will take some time to fix. we will update you on this, once this issue fix. Thanks

**Comment by RamboV** [comment]
Hello @v-sudkharat  made changes accordingly kindly review

**Review by v-sudkharat — APPROVED** [review]
Looks Good. Solution content type Created and Tested successfully. 
Below is a screenshot of deployed and created content type: -
Dashboard

Thanks!

**Review by v-atulyadav — APPROVED** [review]

**Pull request merged**
